snakemake/deployment/conda.py

```diff
@@ -583,24 +635,98 @@ def bin_path(self):
         else:
             return os.path.join(self.prefix_path, "bin")
 
-    def shellcmd(self, env_path, cmd):
+    def shellcmd(self, env_address, cmd):
         # get path to activate script
         activate = os.path.join(self.bin_path(), "activate")
 
         if ON_WINDOWS:
             activate = activate.replace("\\", "/")
-            env_path = env_path.replace("\\", "/")
+            env_address = env_address.replace("\\", "/")
 
-        return "source {} '{}'; {}".format(activate, env_path, cmd)
+        return "source {} '{}'; {}".format(activate, env_address, cmd)
 
-    def shellcmd_win(self, env_path, cmd):
+    def shellcmd_win(self, env_address, cmd):
         """Prepend the windows activate bat script."""
         # get path to activate script
         activate = os.path.join(self.bin_path(), "activate.bat").replace("\\", "/")
-        env_path = env_path.replace("\\", "/")
+        env_address = env_address.replace("\\", "/")
 
-        return '"{}" "{}"&&{}'.format(activate, env_path, cmd)
+        return '"{}" "{}"&&{}'.format(activate, env_address, cmd)
 
 
 def is_mamba_available():
     return shutil.which("mamba") is not None
+
+
+class CondaEnvSpec(ABC):
+    @abstractmethod
+    def apply_wildcards(self, wildcards):
+        ...
+
+    @abstractmethod
+    def get_conda_env(self, workflow, env_dir=None, container_img=None, cleanup=None):
+        ...
+
+    @abstractmethod
+    def check(self):
+        ...
+
+    @property
+    def is_file(self):
+        return False
+
+
+class CondaEnvFileSpec(CondaEnvSpec):
+    def __init__(self, filepath: str, rule=None):
+        if isinstance(filepath, _IOFile):
+            self.file = filepath
+        else:
+            self.file = IOFile(filepath, rule=rule)
+
+    def apply_wildcards(self, wildcards):
+        filepath = self.file.apply_wildcards(wildcards)
+        if is_local_file(filepath):
+            # Normalize 'file:///my/path.yml' to '/my/path.yml'
+            filepath = parse_uri(filepath).uri_path
+        return CondaEnvFileSpec(filepath)
+
+    def check(self):
+        self.file.check()
+
+    def get_conda_env(self, workflow, env_dir=None, container_img=None, cleanup=None):
+        return Env(
+            workflow,
+            env_file=self.file,
+            env_dir=env_dir,
+            container_img=container_img,
+            cleanup=cleanup,
+        )
+
+    @property
+    def is_file(self):
+        return True
+
+
+class CondaEnvNameSpec(CondaEnvSpec):
+    def __init__(self, name: str):
+        self.name = name
+
+    def apply_wildcards(self, wildcards):
+        return CondaEnvNameSpec(apply_wildcards(self.name, wildcards))
```

Comment:
This is really nice to support wildcards in the conda environment name. Do you want to mention that briefly in the docs?